python/tests/integration/server_manager.py

```diff
@@ -103,29 +96,99 @@ def handle_call(self, context, connection):
             if exp.matches(context.message):
                 exp.respond(context, connection)
 
+    def __enter__(self):
+        self.start()
+        return self
+
+    def __exit__(self, *args):
+        self.stop()
+
+    def start(self):
+        assert self.thread is None, 'server already started'
+        self.thread = threading.Thread(target=self.serve)
+        self.thread.start()
+        while not self.ready:
+            pass
```

Comment:
This will spin really fast, taking up 100% CPU for that duration.

Consider using a [Condition](https://docs.python.org/2/library/threading.html#condition-objects) to let the other thread notify this one that it's ready. So, instead of setting `ready` to `True`, that thread can call `ready()` on this class, which will call `notify` on the condition (which this thread was `wait`ing on).